test/e2e/conformance_test.go

```diff
@@ -183,13 +184,23 @@ var _ = Describe("Conformance Tests", func() {
 				LabelSelector: "kubernetes.io/os=linux",
 			}
 
-			noScheduleTaint := &corev1.Taint{
+			// This is kept to support versions of k8s < 1.20.
+			deprecatedNoScheduleTaint := &corev1.Taint{
 				Key:    "node-role.kubernetes.io/master",
 				Value:  "",
 				Effect: "NoSchedule",
 			}
 
-			err := node.TaintNode(workloadProxy.GetClientSet(), options, noScheduleTaint)
+			noScheduleTaint := &corev1.Taint{
```

Comment:
I think we either need to only apply this in 1.24+ or update e2e.test parameter `--NonblockingTaints` to include `node-role.kubernetes.io/control-plane` otherwise jobs < 1.24 will timeout because this new label will be applied.

https://github.com/kubernetes/kubernetes/compare/a37b6fc8b488a0...400b1dea170d9d#diff-c017e0c5e68c14cb966137bc87de3d3ab8475dc25f9b5fdb47a2b8174830054f

Reply:
since the other templates are now adding the flag, my guess is we need to add it to the test.e2e.

Reply:
where do I add the `--NonblockingTaints` flag? 

I changed it to only apply the new taint for 1.24+, would that be enough?

Reply:
The viper config files pass the flags to the `e2e.test` binary: https://github.com/kubernetes-sigs/cluster-api-provider-azure/tree/main/test/e2e/data/kubetest 

see https://github.com/kubernetes-sigs/cluster-api/blob/dd3c2515c3edab63ab08b506c612932ecf3b08f8/test/framework/kubetest/run.go#L169



> I changed it to only apply the new taint for 1.24+, would that be enough?

I can't remember the behavior of the tolerations added to the addons.  If they schedule with those new tolerations added, it would be I believe.

Reply:
looks like conformance is still failing, but I found out the calico deployment also has a `node-role.kubernetes.io/master` toleration so that might be the issue. Going to try adding a patch until we can update calico.